doc/devel/contributing.rst

```diff
@@ -595,3 +595,43 @@ example code you can load it into a file handle with::
 
     import matplotlib.cbook as cbook
     fh = cbook.get_sample_data('mydata.dat')
+
+
+Baseline image generation
+-------------------------
+The first time this test is run, there will be no baseline image to compare against, so the test will fail. The baseline
+images would be generated either by installing ``matplotlib-baseline-images`` package or by running the baseline image
+generation command from the commandline. These methods are described below. Later on, developer can run ::
+
+    python3 -pytest
+
+and the tests will pass.
+
+
+How to install matplotlib-baseline-images package?
+--------------------------------------------------
+Developer can install the ``matplotlib-baseline-images`` package by running the command ::
+
+    python3 -mpip install  -ve  sub-wheels/matplotlib-baseline-images
+
+
+Baseline image generation at the fresh install of matplotlib through command line
+----------------------------------------------------------------------------------
+The baseline images can be generated in the ``baseline_images`` directory by using ::
+
+    python3 -mpytest --generate_images
+
+
+How to generate the modified baseline images?
+---------------------------------------------
+Tests generating the baseline images may change over time. So, the baseline images needs to be regenerated by the
+developer after fetching the changes through ``git pull`` or ``git checkout -b featured_branch``. Missing baseline
+images can be generated in the ``baseline_images`` directory with the following command::
+
+    python3 -mpytest --generate_images
+
+Later on, developer can run the test suite with ::
+
+    python3 -pytest
+
+This time the tests will pass.
```

Comment:
We don't need to assert that the tests will pass (they should always be passing).